src/types.ts

```diff
@@ -70,14 +70,50 @@ export type Amount = {
     multiasset: Multiasset<Uint>,
 }
 
-export type DatumHash = FixlenBuffer<typeof DATUM_HASH_LENGTH>
+// Datum
+export enum DatumType {
+    HASH = 0,
+    INLINE = 1,
+}
 
-export type TransactionOutput = {
+export type DatumHash = {
+    type: DatumType.HASH,
+    hash: FixlenBuffer<typeof DATUM_HASH_LENGTH>,
+}
+
+
+export type DatumInline = {
+    type: DatumType.INLINE,
+    bytes: Buffer,
+}
+
+export type Datum = DatumHash | DatumInline
+
+// Transaction output
+export enum OutputType {
+    LEGACY,
+    POST_ALONZO,
```

Comment:
What about LEGACY and BABBAGE? I find ALONZO quite confusing since the serialization format is much older. If we use LEGACY and BABBAGE, the name corresponds to the time of the format introduction.

Also, @gabrielKerekes suggests including the data structure in the name as a hint for clients: LEGACY_LIST and BABBAGE_MAP. (I have no preference between these two suggestions.)

Reply:
Ah, it's actually ARRAY, not a LIST (from the CBOR standpoint). The public API of Legder/Trezor/serialization lib might represent as stuff as lists or arrays depending on the language, but the essence is how it is serialized. I'm going to change it like that in ledgerjs.